libparsec/crates/types/src/manifest.rs

```diff
@@ -163,7 +163,14 @@ macro_rules! impl_manifest_verify {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
 pub struct BlockAccess {
     pub id: BlockID,
+    /// Offset within the file where the block should be placed.
+    ///
+    /// Note this should not be confused with the offset *within the block* itself.
     pub offset: SizeInt,
+    /// Size of the block.
+    ///
+    /// Note the actual block data size might differ from this value, in which case
+    /// the block should be padded with zeroes.
```

Comment:
Not really: this is always the blocksize except for the last block, and this size should always match the size of the block data.

Reply:
Add a check in `validate_block`
Add a comment in `copy_between_start_and_stop`